modules/memfault-firmware-sdk/Kconfig

```diff
@@ -274,23 +274,33 @@ choice MEMFAULT_HTTP_PERIODIC_UPLOAD_CONTEXT
 	default MEMFAULT_HTTP_PERIODIC_UPLOAD_USE_DEDICATED_WORKQUEUE
 endchoice
 
-# Workaround to select the correct HW ID source
+# Workaround to select the correct HW ID source choice config defaults
 
-if HW_ID_LIBRARY
+if MEMFAULT_NCS_DEVICE_ID_HW_ID || MEMFAULT_NCS_DEVICE_ID_IMEI
 
 choice HW_ID_LIBRARY_SOURCE
-	default HW_ID_LIBRARY_SOURCE_IMEI if MEMFAULT_NCS_DEVICE_ID_IMEI
-	default HW_ID_LIBRARY_SOURCE_NET_MAC if MEMFAULT_NCS_DEVICE_ID_NET_MAC
+	# Default to IMEI if deprecated MEMFAULT_NCS_DEVICE_ID_IMEI is selected, or
+	# for all devices using the modem library
+	default HW_ID_LIBRARY_SOURCE_IMEI if MEMFAULT_NCS_DEVICE_ID_IMEI || NRF_MODEM_LIB
+	# Otherwise, special case for backwards compatibility, use the network MAC on
+	# the nRF7002 DK
+	default HW_ID_LIBRARY_SOURCE_NET_MAC if BOARD_NRF7002DK_NRF5340_CPUAPP
+	# And finally, use the BLE MAC address if Bluetooth is enabled
+	default HW_ID_LIBRARY_SOURCE_BLE_MAC if BT
```

Comment:
Hmm this is actually not ideal under certain conditions- it uses `bt_le_oob_get_local()`, which will generate a new Private Resolvable Address, instead of using the actual private address :(. It may be better to use UID instead, @maxd-nordic can you comment?